share/config.py

```diff
@@ -264,3 +272,107 @@ def parse_config(config_yaml: str) -> Config:
         conf.add_input(current_input)
 
     return conf
+
+
+def aws_sm_expander(config_yaml: str) -> str:
+    needed_secret_arns = {}
+    needed_secret_keys = {}
+
+    re_pattern = r"arn:aws:secretsmanager:(?:[^:]*):(?:[^:]*):secret:(?:[^\n]*)"
+    found_secrets = re.findall(re_pattern, config_yaml)
```

Comment:
nit: suffix `found_secrets` and `secret` etc below with `arn` to clarify what is the exact content